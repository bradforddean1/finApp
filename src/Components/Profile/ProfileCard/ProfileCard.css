.ProfileCard .list {
	display: flex;
	flex-direction: column;
}
