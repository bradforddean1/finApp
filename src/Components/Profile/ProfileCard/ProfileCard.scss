@use "../../../scss/variables" as v;

.ProfileCard .list {
	display: flex;
	flex-direction: column;
}

.ProfileCard h3.card-line {
	color: v.$accent-color;
}
