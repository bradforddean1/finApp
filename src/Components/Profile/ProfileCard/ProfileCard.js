import React from "react";
import "./ProfileCard.css";
import PropTypes from "prop-types";
import Line from "../ProfileCardLine/ProfileCardLine";

/**
 * Data card for security profile page
 * @component
 */
function ProfileCard(props) {
	const { title, metrics, values } = props;
	return (
		<div className="ProfileCard card bg-secondary">
			<h3 className="card-line accent">{title}</h3>
			<ul className="list">
				{metrics.map((metric, i) => (
					<Line key={i} name={metric} value={values[i]} />
				))}
			</ul>
		</div>
	);
}

ProfileCard.propTypes = {
	/**
	 * Header subject name
	 */
	title: PropTypes.string.isRequired,
	/**
	 * Array of metrics to be inluded on card.
	 */
	metrics: PropTypes.arrayOf(PropTypes.object),
};

ProfileCard.defaultProps = {
	metrics: [],
};

export default ProfileCard;
