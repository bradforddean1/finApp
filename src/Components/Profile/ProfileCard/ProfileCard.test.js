import React from 'react';
import ReactDOM from 'react-dom';
import ProfileCard from './ProfileCard';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('ProfileCard Component', () => {
    it('Renders',
        () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <ProfileCard />,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it.skip('Renders the Default State',
    () => {
        const wrapper = shallow(<ProfileCard />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it.skip('Renders given props',
    () => {
        const wrapper = shallow(<ProfileCard {...props}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
