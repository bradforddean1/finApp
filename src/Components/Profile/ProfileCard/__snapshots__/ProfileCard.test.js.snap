// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileCard Component Renders the Default State 1`] = `
<div
  className="ProfileCard card bg-secondary"
>
  <h3
    className="card-line"
  />
  <ul
    className="list"
  />
</div>
`;
